core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -259,6 +352,19 @@ export class ResultPanelComponent {
 
     // generate columnDef from first row, column definition is in order
     this.currentColumns = ResultPanelComponent.generateColumns(columns);
+    this.total = totalRowCount ?? resultData.length;
+
+    // get the current page size, if the result length is less than `this.currentPageSize`,
+    //  then the maximum number of items each page will be the length of the result, otherwise `this.currentPageSize`.
+    this.currentPageSize = Math.min(this.total, this.currentPageSize);
+
+    // save paginated result into session storage
+    sessionStorage.setItem('newWorkflowExecuted', 'false');
+    sessionStorage.setItem('currentResult', JSON.stringify(this.currentResult));
+    sessionStorage.setItem('currentPageIndex', JSON.stringify(this.currentPageIndex));
+    sessionStorage.setItem('currentPageSize', JSON.stringify(this.currentPageSize));
+    sessionStorage.setItem('total', JSON.stringify(this.total));
+    sessionStorage.setItem('columnKeys', JSON.stringify(columnKeys));
```

Comment:
see the sessionstorage comment above

Reply:
Updated in https://github.com/Texera/texera/pull/903/commits/d668ef2d6eb6c5ef694c00eeaa984ccf51bd860e